cnf-certification-test/networking/suite.go

```diff
@@ -67,8 +80,106 @@ var _ = ginkgo.Describe(common.NetworkingTestKey, func() {
 	ginkgo.It(testID, ginkgo.Label(testID), func() {
 		testMultusNetworkConnectivity(&env, defaultNumPings, netcommons.IPv6)
 	})
+	// Default interface ICMP IPv6 test case
+	testID = identifiers.XformToGinkgoItIdentifier(identifiers.TestUndeclaredContainerPortsUsage)
+	ginkgo.It(testID, ginkgo.Label(testID), func() {
+		testListenAndDeclared(&env)
+	})
 })
 
+func parseListening(res string, listeningPorts map[key]string) {
+	var k key
+	lines := strings.Split(res, "\n")
+	for _, line := range lines {
+		fields := strings.Fields(line)
+		if !strings.Contains(line, "LISTEN") {
+			continue
+		}
+		if indexprotocolname > len(fields) || indexport > len(fields) {
+			return
+		}
+		s := strings.Split(fields[indexport], ":")
+		p, _ := strconv.Atoi(s[1])
+		k.port = p
+		k.protocol = strings.ToUpper(fields[indexprotocolname])
+		k.protocol = strings.ReplaceAll(k.protocol, "\"", "")
+		listeningPorts[k] = ""
+	}
+}
+
+func checkIfListenIsDeclared(listeningPorts, declaredPorts map[key]string) map[key]string {
+	res := make(map[key]string)
+	if len(listeningPorts) == 0 {
+		return res
+	}
+	for k := range listeningPorts {
+		_, ok := declaredPorts[k]
+		if !ok {
+			tnf.ClaimFilePrintf(fmt.Sprintf("The port %d on protocol %s in pod %s is not declared.", k.port, k.protocol, listeningPorts[k]))
+			res[k] = listeningPorts[k]
+		}
+	}
+	return res
+}
+
+//nolint:funlen
+func testListenAndDeclared(env *provider.TestEnvironment) {
+	var k key
+	declaredPorts := make(map[key]string)
+	listeningPorts := make(map[key]string)
+	var failedPods string
+	var skippedPods string
+
+	for _, pod := range env.Pods {
+		for i := 0; i < len(pod.Spec.Containers); i++ {
+			container := &pod.Spec.Containers[i]
+			ports := container.Ports
+			fmt.Println(ports)
+			if ports == nil {
+				tnf.ClaimFilePrintf("Failed to get declared port for container %d due to %v, skipping pod %s", container, pod.Namespace+"."+pod.Name)
+				skippedPods += pod.Namespace + " " + pod.Name + " " + container.Name + "\n"
+			}
+			for i := 0; i < len(ports); i++ {
+				k.port = int(ports[i].ContainerPort)
+				k.protocol = string(ports[i].Protocol)
+				declaredPorts[k] = ports[i].Name
+			}
+		}
+
+		oc := clientsholder.GetClientsHolder()
+		output, outerr, err := oc.ExecCommandContainer(clientsholder.Context{Namespace: pod.Namespace,
+			Podname: pod.Name, Containername: pod.Spec.Containers[0].Name}, `ss -tulwnH`)
```

Comment:
Here you're getting the listening ports of the first container of that pod... what about the other containers in the same pod? Am I missing something?

Reply:
Yeah, I also think that this code should be moved up inside the containers loop

Reply:
The old repo has a similar implementation issue in this TC, so I think it should be changed there as well.

Reply:
yes, I agree. and I fixed it.